docs/proposals-done/202203-grpc-query-api.md

```diff
@@ -0,0 +1,108 @@
+## Introduce a Query gRPC API
+
+* **Owners:**
+    * `@fpetkovski` 
+    * `@mzardab` 
+
+> TL;DR: Introducing a new gRPC API for `/query` and `/query_range`
+
+## Why
+We want to be able to distinguish between gRPC Store APIs and other Queriers in the query path. Currently, Thanos Query implements the gRPC Store API and the root Querier does not distinguish between Store targets and other Queriers that are capable of processing a PromQL expression before returning the result. The new gRPC Query API will allow a querier to fan out query execution, in addition to Store API selects.
```

Comment:
> We want to be able to distinguish between gRPC Store APIs and other Queriers in the query path.

I don't get that - you can distinguish it easily with configuration, no need for a new API for this particular point?

Reply:
Sure, but we can't distinguish between them via gRPC, only via HTTP.